auth/src/main/java/org/aerogear/mobile/security/checks/ScreenLockCheck.java

```diff
@@ -0,0 +1,31 @@
+package org.aerogear.mobile.security.checks;
+
+import android.app.KeyguardManager;
+import android.content.Context;
+import android.os.Build;
+
+import org.aerogear.mobile.security.SecurityCheck;
+import org.aerogear.mobile.security.SecurityCheckResult;
+import org.aerogear.mobile.security.impl.SecurityCheckResultImpl;
+
+/**
+ * A check for whether the device the application is running on has a screen lock.
+ */
+public class ScreenLockCheck implements SecurityCheck {
+    private static final String NAME = "detectScreenLock";
+
+    /**
+     * Check whether the device has an automatic screenlock
+     * @param context Context to be used by the check.
+     * @return <code>true</code> if the device has an automatic screen lock
+     */
+    @Override
+    public SecurityCheckResult test(Context context){
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M){
+            KeyguardManager keyguardManager = (KeyguardManager) context.getSystemService(Context.KEYGUARD_SERVICE);
+            return new SecurityCheckResultImpl(NAME, !keyguardManager.isDeviceSecure());
+        }
+        return new SecurityCheckResultImpl(NAME, false);
```

Comment:
so if the devices is running on android API version pre-M, this always return false?

Not sure if this is the right behaviour. I think for pre-M, we can use [isKeyguardSecure](https://developer.android.com/reference/android/app/KeyguardManager.html#isKeyguardSecure())